File: ABC081B.js
// inputに入力データ全体が入る
function Main(input) {
	input = input.split('\n');
	var n = input[0];
	var tmp = input[1].split(' ');
  	var isEven = true;
  	var count = 0;
	while(isEven){
        for(var i = 0; i < n; i++){
          if(tmp[i]%2 == 1){
            isEven = false;
          }
          else{
            tmp[i] /= 2;
          }
    	}
      	if(isEven){
          count++;
        }
    }
  	console.log(count);
}
//*この行以降は編集しないでください（標準入出力から一度に読み込み、Mainを呼び出します）
Main(require("fs").readFileSync("/dev/stdin", "utf8"));

File: ABC081B.py
N = input()
A = list(map(int, input().split()))

isEven = True
count = 0
while isEven:
    for i in A:
        if i % 2 == 1:
            isEven = False
    if isEven:
        A = list(map(lambda x: x/2, A))
        count += 1
print(count)
